tests/rptest/tests/acls_test.py

```diff
@@ -126,6 +126,8 @@ def prepare_cluster(self,
         if self.security.sasl_enabled() or enable_authz:
             self.admin.create_user("cluster_describe", self.password,
                                    self.algorithm)
+
+        self.admin.create_user(*self.redpanda.SUPERUSER_CREDENTIALS)
         client = self.get_super_client()
```

Comment:
> There is a flaky failure on CDT where we fail to create the superuser
before running our permission checks.

ok but can you describe what the problem is? this looks correct, but it also looks like it should have never worked before, but that seems to also not be the case.

Reply:
> > There is a flaky failure on CDT where we fail to create the superuser
> > before running our permission checks.
> 
> ok but can you describe what the problem is? this looks correct, but it also looks like it should have never worked before, but that seems to not be the case.

My reasoning is in the ticket
https://github.com/redpanda-data/redpanda/issues/7536#issuecomment-1335847144

We do create the admin user in RedpandaService but, based on the logs, it looks like the user was not created in time.

Reply:
> We do create the admin user in RedpandaService but, based on the logs, it looks like the user was not created in time.

as test, if there is a contract with RedpandaService that it will provide a cluster with a superuser for the test to utilize, but that sometimes breaks, we should address that rather than creating the user in this one specific test.

from the logs do you see that the redpanda service fails to create the user? i could see us needing to add a retry to that creation. the other scenario might be that there is a path in service start-up that skips super user creation.

Reply:
> > We do create the admin user in RedpandaService but, based on the logs, it looks like the user was not created in time.
> 
> @NyaliaLui as test, if there is a contract with RedpandaService that it will provide a cluster with a superuser for the test to utilize, but that sometimes breaks, we should address that rather than creating the user in this one specific test. does that contract exist?
> 
> from the logs do you see that the redpanda service fails to create the user? i could see us needing to add a retry to that creation. the other scenario might be that there is a path in service start-up that skips super user creation.

Yes, the call to `admin.create_user` in RedpandaService triggered, and **all** RP brokers failed to create the user

Debug log from ducktae
```
[INFO  - 2022-11-27 12:55:22,019 - admin - create_user - lineno:609]: Creating user admin:admin:SCRAM-SHA-256
``` 

Logs from the x3 brokers
```
WARN  2022-11-27 12:55:22,022 [shard 0] request_auth - request_auth.cc:107 - Client auth failure: user '{admin}' not found

WARN  2022-11-27 12:55:22,028 [shard 0] request_auth - request_auth.cc:107 - Client auth failure: user '{admin}' not found

WARN  2022-11-27 12:55:22,147 [shard 0] request_auth - request_auth.cc:107 - Client auth failure: user '{admin}' not found
```